v2/tools/generator/internal/astmodel/local_package_reference.go

```diff
@@ -109,6 +109,20 @@ func (pr LocalPackageReference) ApiVersion() string {
 	return pr.apiVersion
 }
 
+// CreateImportAlias creates a custom alias for importing this reference
+func (pr LocalPackageReference) CreateImportAlias(style PackageImportStyle) string {
```

Comment:
It's a bit weird that this is a function on `LocalPackageReference`, because all of the `PackageReference`'s are really just the _reference_ part of things, not the name (the name is `PackageImport`). In many contexts that a `PackageReference` is used (such as for describing the object graph), this method doesn't really make sense.

Would it be cleaner to instead put this on `PackageImport`? I suppose that'd also involve moving the helper methods for `SimplifiedXyz`, but I think they could be written in terms of the Getters on this type anyway.

Only suggesting this because it feels like the name generation/simplification logic should be on the type with the name (`PackageImport`) and not on the type that's just the "link"

Reply:
I was 50/50 on putting the logic there originally - it was largely to leverage polymorphism, but you've convinced me to move them.